Fixes #9412: Context command ignored implicit binders (#11390)
coq/coq: doc/sphinx/language/core/sections.rst

.. cmd:: Context {+ @binder }

>    When in a section, this combines several instances of
>    :cmd:`Variable` or :cmd:`Let` at once, also supporting
>    :ref:`implicit-generalization`. Variables which are surrounded by
>    %{ %} are turned into implicit arguments at the end of section.
> 
>    When not in a section, this combines several instances of
>    :cmd:`Local Parameter` or :cmd:`Local Definition` at once, also
>    supporting :ref:`implicit-generalization`.

Reviewer: ```suggestion
   Inside sections, combines several instances of
   :cmd:`Variable` or :cmd:`Let`, also supporting
   :ref:`implicit-generalization`. Variables which are surrounded by
   %{ %} are turned into implicit arguments at the end of section.

   Outside of section, combines several instances of
   :cmd:`Local Parameter` or :cmd:`Local Definition`, also
   supporting :ref:`implicit-generalization`.
```

But I have no idea what "combining" means and where the result of such combining would end up.  The "Declare variables" language previously used makes more sense (though would need to be adjusted to the inside/outside differences).

Is it also true that Context within a section is local to the section?

Reviewer 2: I agree that the new description is more confusing than the previous one (which was mostly fine except that it didn't say what happened outside sections). But if we really want to highlight the use of `Context` outside sections, we should move this command out of the page on sections and next to the one of `Parameter` and `Variable`. To be discussed with the possible generalization of the `Parameter` and `Variable` syntax and also in relation to #13068. In any case, not really relevant to this PR.

>Is it also true that Context within a section is local to the section?

Yes.
